crates/codegen/schema/src/types/parser.rs

```diff
@@ -0,0 +1,66 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+pub type ParserRef = std::rc::Rc<Parser>;
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash)]
+#[serde(deny_unknown_fields)]
+pub struct Parser {
+    #[serde(default)]
+    pub name: Option<String>,
+    #[serde(default)]
+    pub lookahead: Option<ParserRef>,
+    #[serde(flatten)]
+    pub definition: ParserDefinition,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub enum ParserDefinition {
+    #[schemars(title = "Choice Expression")]
```

Comment:
```suggestion
    #[schemars(title = "Choice Parser Definition")]
```

I see this PR renames "expressions" in the schema to "definitions". But there are many other locations in codegen crates that still use `expression` or `expr`. Should we be renaming them as well?

Reply:
Same naming issue.

Reply:
* https://github.com/NomicFoundation/slang/issues/373